File: BlueJ_Proyecto/Bitacora.java
import java.util.ArrayList;
import java.util.Date;

/**
 * Historial clinico de eventos de un ticket.
 */
public class Bitacora
{
    private ArrayList<String> registros;

    public Bitacora()
    {
        registros = new ArrayList<>();
    }

    public void registrarEvento(String evento)
    {
        registros.add("[" + new Date() + "] " + evento);
    }

    public ArrayList<String> getRegistros()
    {
        return registros;
    }

    public void mostrarBitacora()
    {
        System.out.println("  Historial clinico:");
        if (registros.isEmpty())
        {
            System.out.println("    (sin eventos registrados)");
            return;
        }
        for (String evento : registros)
        {
            System.out.println("    - " + evento);
        }
    }
}

File: ClinicaVeterinaria/logica/Bitacora.cs
using System;
using System.Collections.Generic;

namespace ClinicaVeterinaria.logica
{
    public class Bitacora
    {
        public List<string> lstRegistros { get; set; }

        public Bitacora()
        {
            lstRegistros = new List<string>();
        }

        public void registrarEvento(string strEvento)
        {
            lstRegistros.Add($"{DateTime.Now:dd/MM/yyyy HH:mm} - {strEvento}");
        }

        public void mostrarBitacora()
        {
            Console.WriteLine("  Historial clinico:");
            if (lstRegistros.Count == 0)
            {
                Console.WriteLine("    (sin eventos registrados)");
                return;
            }
            foreach (var strEvento in lstRegistros)
                Console.WriteLine($"    - {strEvento}");
        }
    }
}
